cli/src/commands/cloud/environments/configure.rs

```diff
@@ -0,0 +1,250 @@
+    type Err = std::convert::Infallible;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s.split_once('/') {
+            Some((account, environment)) => Ok(Self(Some(account.into()), environment.into())),
+            None => Ok(Self(None, s.into())),
+        }
+    }
+}
+
+pub async fn run_configure(State(env): State<CliEnv>, opts: &Configure) -> Result<()> {
+    let config_data = if env.config_file.is_file() {
+        std::fs::read_to_string(env.config_file.as_path())?
+    } else {
+        "".into()
+    };
+
+    let mut doc = config_data
+        .parse::<DocumentMut>()
+        .context("Failed to parse config file as TOML")?;
+
+    let client = CloudClient::new(&env)?;
+
+    let accounts = with_progress("Listing accounts...", async {
+        client.list_accounts().await?.into_body().await
+    })
+    .await?
+    .accounts;
+
+    let (account, environment) =
+        if let Some(EnvironmentSpecifier(account, environment)) = &opts.environment_specifier {
+            (
+                account.as_ref().map(String::as_str),
+                Some(environment.as_str()),
+            )
+        } else {
+            (None, None)
+        };
+
+    let account_i = if let Some(account) = account {
+        match accounts
+            .iter()
+            .find_position(|acc| acc.account_id == account)
+            .or(accounts.iter().find_position(|env| env.name == account))
+        {
+            Some((i, _)) => i,
+            None => return Err(anyhow::anyhow!("Couldn't find account {account}",)),
+        }
+    } else {
+        match accounts.len() {
+            0 => {
+                return Err(anyhow::anyhow!(
+                    "No accounts set up; use the Restate Cloud UI to create your first account"
+                ))
+            }
+            1 => 0,
+            _ => account_picker(&accounts)?,
+        }
+    };
+
+    let environments = with_progress("Listing environments...", async {
+        client
+            .list_environments(&accounts[account_i].account_id)
+            .await?
+            .into_body()
+            .await
+    })
+    .await?
+    .environments;
+
+    let environment_i = if let Some(environment) = environment {
+        match environments
+            .iter()
+            .find_position(|env| env.environment_id == environment)
+            .or(environments
+                .iter()
+                .find_position(|env| env.name == environment))
+        {
+            Some((i, _)) => i,
+            None => {
+                return Err(anyhow::anyhow!(
+                    "Couldn't find environment {environment} in account {}",
+                    accounts[account_i].name,
+                ))
+            }
+        }
+    } else {
+        match environments.len() {
+            0 => {
+                return Err(anyhow::anyhow!(
+                "No environments set up; use the Restate Cloud UI to create your first environment"
```

Comment:
Do you think we should return this as an anyhow error, or should we take control over the formatting and eprintln instead?